opentrons-ai-server/api/domain/openai_predict.py

```diff
@@ -1,25 +1,211 @@
+            "Ω": "O",
+        }
+        translation_table = str.maketrans(greek_to_english)
+        translated_text = prompt.translate(translation_table)
+        return translated_text
+
+    def get_docs_all(self, query: str) -> Tuple[str, str, str]:
+        commands = self.extract_atomic_description(query)
+        print(f"commands: {commands}")
+
+        # define file paths for storage
+        example_command_path = str(ROOT_PATH / "api" / "storage" / "index" / "commands")
+        documentation_path = str(ROOT_PATH / "api" / "storage" / "index" / "v215")
+        labware_api_path = standard_labware_api
+
+        # retrieve example commands
+        example_commands = f"\n\n{'='*15} EXAMPLE COMMANDS {'='*15}\n"
+        storage_context = StorageContext.from_defaults(persist_dir=example_command_path)
+        index = load_index_from_storage(storage_context)
+        retriever = index.as_retriever(similarity_top_k=1)
+        content_all = ""
+        if isinstance(commands, list):
+            for command in commands:
+                nodes = retriever.retrieve(command)
+                content = "\n".join(node.text for node in nodes)
+                content_all += f">>>> >>>> \n\\{content}n"
+            example_commands += content_all
+        else:
+            example_commands = []
+
+        # retrieve documentation
+        storage_context = StorageContext.from_defaults(persist_dir=documentation_path)
+        index = load_index_from_storage(storage_context)
+        retriever = index.as_retriever(similarity_top_k=3)
+        nodes = retriever.retrieve(query)
+        docs = "\n".join(node.text.strip() for node in nodes)
+        docs_v215 = f"\n{'='*15} DOCUMENTATION {'='*15}\n\n" + docs
+
+        # standard api names
+        standard_api_names = f"\n{'='*15} STANDARD API NAMES {'='*15}\n\n" + labware_api_path
+
+        return example_commands, docs_v215, standard_api_names
+
+    def extract_atomic_description(self, protocol_description: str) -> List[str]:
+        class atomic_descr(BaseModel):
+            """
+            Model for atomic descriptions
+            """
+            desc: List[str]
+
+        program = OpenAIPydanticProgram.from_defaults(
+            output_cls=atomic_descr, 
+            prompt_template_str=prompt_template_str.format(protocol_description=protocol_description), 
+            verbose=False, 
+            llm=liOpenAI(model="gpt-4-1106-preview")
+        )
+        details = program(protocol_description=protocol_description)
+        descriptions = []
+        print("=" * 50)
+        for x in details.desc:
+            if x not in ["Modules:", "Adapter:", "Labware:", "Pipette mount:", "Commands:", "Well Allocation:", "No modules"]:
+                descriptions.append(x)
+        return descriptions
+
+    def refine_response(self, assitant_message: str) -> str:
+        if assitant_message is None:
+            return ""
+        sys_msg: ChatCompletionMessageParam = {
```

Comment:
for consistency L177: user_message
```suggestion
        sys_message: ChatCompletionMessageParam = {
```

or `system_message`

Reply:
Done. Changed to `system_message`